CodeDay-ProductivityApp/codeDay-productivity
# Path: /users/{id}/goals/{id}

closes #7 

At the path `/users/{id}/goals/{id}` we want to preform read (get), update (put), and delete functionality on a goal.
This path gives us the goal id to access a single goal.

The get functionality returns the goal, as of now it returns a null if the id does not exist.
The put functionality returns the goal, throws a 404 error if the id does not exist.
The delete functionality returns the goal, as of now it returns a null if the id does not exist.

To test:
run the app
using Postman:
-create at least one User
-create a couple of Goals for the User
at localhost:8080/api/v1/users/{you user id}/goals/{your goal id}:
-use a Get call to retrieve the Goal
-use a Put call with a Header of Content-Type with a Value of application/json and a Body with a Raw JSON object containing the property or properties you want to update for the Goal
-use a Delete call to delete the Goal

The update logic is clunky, there is most likely a better way to map the changes to the Goal. There is built in error handling, not sure on best procedures here. As there wasn't any written testing for the work Nahom pushed, I wasn't sure about the best way to add testing for the code I added. There should probably be an issue created for adding testing for the code that is already on the main branch.
